src/ResourceManager/Resources/Commands.Resources/help/Get-AzureRmPolicyAssignment.md

```diff
@@ -13,22 +13,43 @@ Gets policy assignments.
 
 ## SYNTAX
 
-### GetAllPolicyAssignments (Default)
+### DefaultParameterSet (Default)
 ```
 Get-AzureRmPolicyAssignment [-ApiVersion <String>] [-Pre] [-DefaultProfile <IAzureContextContainer>]
  [-InformationAction <ActionPreference>] [-InformationVariable <String>] [<CommonParameters>]
 ```
 
-### GetPolicyAssignmentName
+### NameParameterSet
 ```
-Get-AzureRmPolicyAssignment [-Name <String>] -Scope <String> [-PolicyDefinitionId <String>]
```

Comment:
It's not worth the work of obsoleting it. I'll just bring it back in, and it will remain a no-op.

Reply:
It turns out that I can't bring that parameter back in without removing some new functionality I added. This update adds the ability to give -Name without -Scope (previously that was not allowed). In order to return -PolicyDefinitionId to the way it was, I need to remove that functionality. So to avoid a breaking change to a combination of parameters that doesn't make sense (and doesn't even work), I have to actually put back a bug that was present in preview cmdlets and remove new functionality from the GA cmdlets. I don't see that as a win for the customer. Given the situation, would you reconsider this point?

Reply:
What was the previous behavior of the cmdlet if the user provided the `-PolicyDefinitionId` parameter? If it's the case that the cmdlet would throw an exception everytime, then it is fine to remove this parameter since no user would have an existing script with this parameter that worked properly. If it didn't throw an exception (even if its value wasn't being used), then it's a breaking change, which is something that we cannot allow until our next breaking change release. I agree that exposing new functionality to customers is a much better experience than keeping functionality that doesn't work as intended, but our goal is to avoid making breaking changes.

Looking at the changes and additions you are making to the parameter sets, it seems that you can bring back the `-PolicyDefinitionId` parameter to the `NameParameterSet` (to avoid the breaking change) and remove the `-Scope` parameter from the `PolicyDefinitionIdFilterParameterSet` set to keep the new functionality (the `-PolicyDefinitionId` + `-Scope` scenario should now be handled in the `NameParameterSet` set). This should allow you to handle all previous cases as well as the new cases you are introducing. Do you see any issue with this?

Reply:
Yeah, that's what I thought too, but that combination doesn't allow -PolicyDefinitionId with no other parameters, which needs to be valid input.

Reply:
The previous behavior was that the value for -PolicyDefinitionId was silently ignored, so the results back were the same whether you provided -PolicyDefinitionId or not, and no matter what nonsense value you might have provided.

Reply:
Are we able to remove the `PolicyDefinitionIdFilterParameterSet` set all together and bring back the `-PolicyDefinitionId` parameter to the `NameParameterSet` as an optional parameter? That allows you to provide `-PolicyDefinitionId` with no other parameters, and if either of `-Scope` or `-Name` are provided, you can ignore the value of `-PolicyDefinitionId` like you did previously

Reply:
No. PolicyDefinitionId has a parameter set because it needs to be mutually exclusive from -All and -Effective. Even after removing -Effective, it still needs to be exclusive from -All (or -IncludeDescendents if it gets renamed). I could enforce that by explicit check instead of PS parameter sets, but I'm guessing you won't like that approach.